### PR TITLE
[3.13] gh-124442: make `__static_attributes__` deterministic by sorting (GH-124492)

### PR DESCRIPTION
**Pull request by JelleZijlstra**
(cherry picked from commit 04c837d9d8a474777ef9c1412fbba14f0682366c)

* Issue: gh-124442

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
@Yhg1s it's probably good to get this into the next RC. We can do it later too (https://github.com/python/cpython/issues/124442#issuecomment-2371996974), but would be nicer to get this right the first time.

File: Python/compile.c [hunk 1]
```diff
@@ -2544,7 +2544,16 @@ compiler_class_body(struct compiler *c, stmt_ty s, int firstlineno)
         return ERROR;
     }
     assert(c->u->u_static_attributes);
-    PyObject *static_attributes = PySequence_Tuple(c->u->u_static_attributes);
+    PyObject *static_attributes_unsorted = PySequence_List(c->u->u_static_attributes);
+    if (static_attributes_unsorted == NULL) {
+        return ERROR;
```
Comment (Yhg1s) [inline Python/compile.c:2550]: Shouldn't these early exits call `compiler_exit_scope(c)` like the existing early exit does?

Comment (JelleZijlstra) [reply Python/compile.c:2550]: Yes, good point. Will send a PR to main too.

Comment (JelleZijlstra) [reply Python/compile.c:2550]:
```suggestion
        compiler_exit_scope(c);
        return ERROR;
```

File: Python/compile.c [hunk 2]
```diff
@@ -2544,7 +2544,16 @@ compiler_class_body(struct compiler *c, stmt_ty s, int firstlineno)
...
+    }
+    if (PyList_Sort(static_attributes_unsorted) != 0) {
+        Py_DECREF(static_attributes_unsorted);
```
Comment (JelleZijlstra) [inline Python/compile.c:2554]:
```suggestion
        compiler_exit_scope(c);
        Py_DECREF(static_attributes_unsorted);
```

File: Python/compile.c (hunk 1 above)
Comment (JelleZijlstra) [reply Python/compile.c:2550]: No change needed on main; the code is quite different there. Sorry for missing this while creating the backport.

**Pull request merged**